app/src/org/commcare/preferences/DeveloperPreferences.java

```diff
@@ -120,6 +121,19 @@ protected int getPreferencesResource() {
     public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
         super.onCreatePreferences(savedInstanceState, rootKey);
         setHasOptionsMenu(true);
+        EditTextPreference editTextPreference = (EditTextPreference)findPreference(PREFS_LOG_POST_URL_KEY);
+        if (editTextPreference != null) {
+            editTextPreference.setOnPreferenceChangeListener((preference, newValue) -> {
+                if (newValue == null || newValue.toString().trim().isEmpty()
+                        || newValue.equals(getString(R.string.logs_server_url_protocol))) {
+                    SharedPreferences appPreferences = CommCareApplication.instance().getCurrentApp()
+                            .getAppPreferences();
+                    appPreferences.edit().remove(PREFS_LOG_POST_URL_KEY).apply();
```

Comment:
think we should be defaulting to the url from app profile xml instead of completely removing it.

Reply:
I think this is related to the notion of what is the default URL for the logs submission, right now it seems to be the form submission URL

Reply:
yeah we should be defaulting it to form submission url instead of removing it completely.

Reply:
@shubham1g5 that would be changing the way this preference is currently used and create redundancy. Right now, it seems that we only set the preference when the logs URL is different from the form submissions URL. Defaulting to the submissions URL and to keep consistency, would require setting it during app installation as well, right?

Reply:
@shubham1g5 that was my first impulse as well, but I think it's actually too dangerous to set this redundantly for two reasons.

1 - It introduces a new parity requirement. If we changed the submission URL for form submissions on HQ's end, we'd now need to **also** change the log URL, but we can't really do that with HQ settings because it'll blow away the config for anyone who's added the custom setting.
2 - From a usability perspective, it's dangerous because the users don't have a super meaningful way currently to know what **was** in the box for this current UI before. If the Log URL is Empty when you load it, it's clear that setting it back to blank should be the "inverse" that sets things back to neutral. If you load up the field then and its got the HQ default form URL already in it, I think you'd naturally then assume that you need to keep track of that URL so you can set it back if you want to undo the action.

Reply:
> Defaulting to the submissions URL and to keep consistency, would require setting it during app installation as well, right?

Makes sense, I didn't realise we don't do it already. 

> we can't really do that with HQ settings because it'll blow away the config for anyone who's added the custom setting.

I am not sure I understand it. My thought was to do this as a one time action at app install time -

1. While Installing app, we populate the `logs_receiver_url` in app preferences by copying it from form submission url.

2. When user opens the logs url preference to edit, they would see what this url is pointing to (which would be same as form submission url)

3. User edits the url to some other valid url. We preserve it and show it to user in this preference

4. User edits the url and change is to a non-valid url (empty or something else) -> We show user an error that the change they made is invalid and preserve whatever the old value of the url was. 


No strong thoughts here though overall, I think we want to preserve the existing behaviour in the PR, probably better to rename the field from `Log Submission Server` to `Custom Log Submission Server` to make it more clear that existing log server is not empty when they see it empty.

Reply:
@shubham1g5 I think it's a matter of interpretation, my understanding from going through the code is that we don't have a `Log Submission Server` per se, everything is to be sent to the `Form Submission Server`. However, we can have the option to set a `Log Submission Server`. Is this a valid interpretation? I'm happy to follow the `Custom Log Submission Server` approach

Reply:
Agreed with "Custom" and updated suggestions on how to be consistent with my take on this above. I think it'd introduce a lot of ambiguity to create a new situation where we **expect** the log submission URL to be set, rather than treating this as an override. 

@shubham1g5 I think the big drawback in your user story is that currently users have an expectation that this URL is synchronized, and the 'custom' version is a rare edge case, so I'd be concerned about situations where we'd need to keep the two in sync across other events or changes by default.